### ISSUE TITLE
Timeout switching modes is too short to install an app

### ISSUE THREAD

**Issue by AndreMPCosta** [ma1co/Sony-PMCA-RE#119]
Camera takes a while to switch modes:

```
pmca-console-v0.17-win.exe install -i
Loading app list
Found 10 apps
 [ 1] com.github.ma1co.openmemories.tweak
 [ 2] com.jonasjuffinger.timelapse
 [ 3] com.github.ma1co.openmemories.appstore
 [ 4] com.obsidium.focusbracket
 [ 5] com.obsidium.bettermanual
 [ 6] org.bostwickenator.dofmath
 [ 7] org.bostwickenator.googlephotos
 [ 8] com.github.ma1co.pmcademo.app
 [ 9] org.bostwickenator.ftpuploader
 [10] com.github.LubikR.synologyuploader
Enter number of app to install (0 to abort): 2

Installing com.jonasjuffinger.timelapse
Using drivers Windows-MSC, Windows-MTP
Looking for Sony devices

Querying mass storage device
Sony DSC is a camera in mass storage mode

Switching to app install mode
Waiting for camera to switch...
Operation timed out. Please run this command again when your camera has connected.
```

I can't do anything past this, if I reset the camera to MTP mode the same thing happens, if I try to install while in this mode, it says the camera isn't connected, if the timeout was longer it probably would work.

**Comment by lorient7** [comment]
I have the same problem.
I have tried with all the versions of pmca, and with different versions of Windows and MacOS.
I have a Sony FDR-AX33 camera and it always gives me the error "Operation time out".

**Comment by ma1co** [comment]
The problem isn't the timeout. As the output says, you can simply re-run the command when the camera is connected, it will be detected. The problem is that your camera doesn't connect. What is your camera model? Does it display the app installation screen?

**Comment by AndreMPCosta** [comment]
As you can see in the output "Querying mass storage device
Sony DSC is a camera in mass storage mode"

The camera successfully connects, it just takes a while to switch to app installation mode. The timeout is just like 5 seconds, it's not enough to switch modes.

**Comment by ma1co** [comment]
In that case, just run the command again as soon as the camera has switched.

**Comment by ma1co** [comment]
No, the program always searches both in mass storage and mtp mode. As I wrote above, the problem is not the timeout value, it's that your camera is not detected.

**Comment by AndreMPCosta** [comment]
So enlighten me why the camera magically switches to app installation mode. Care to explain this magic, if the camera, as you say, is not detected.

**Comment by lorient7** [comment]
I have the same problem with the Sony FDR-AX33 (PAL version) camera.
I have tried in all ways and with all versions of the pmca application and several operating systems (Win10, Win7, MacOS). And I always receive the error of time out.
I have tried with the USB mass storage mode and the MTP.
I have tried several times to install the application, just when I get the "time out" error, and it does not work, it keeps giving the error.
I have tried with the web installation, and it gives me a connection error.

 On the other hand, when I ask for information from the camera, I am given correctly. And when I run the option start tweaking (updater mode) it does not give me an error either, but the option to switch between PAL and NTSC does not appear on the camera.
I have tried to install an apk, and it also gives me time out error.

**Comment by ma1co** [comment]
@AndreMPCosta The camera switches from mass storage mode to app install mode (MTP). This means that it shows up as a new usb device. The computer / program doesn't detect the mtp device, so the operation times out. If you re-run the command, the device is still not detected, so you get the same result. So once again: What is your camera model?

@lorient7 Have you enabled usb app installation using tweaking in updater mode? Regarding the error when disabling the language tweak, please open a new issue and post the error message.

**Comment by lorient7** [comment]
> Have you enabled usb app installation using tweaking in updater mode? Regarding the error when disabling the language tweak, please open a new issue and post the error message.**

---------------------------------

Yes, I have enabled usb app installation using tewaking in updater mode. I have enabled all options (languages, pal / ntsc etc). And I have tried combinations of all options. That's when I detected that I could not deselect the language section.

I have exactly the same problem as AndreMPCosta. Pmca detects my camera, and I can ask for information from the camera and show it to me. I can also use tewaking update mode. But when I try to install an application of the 10 available, or when I try to install an application by selecting the apk file, I get the following error:

Switching to app install mode
Waiting for camera to switch...
Operation timed out. Please run this command again when your camera has connected.

**Comment by voidism** [comment]
The problem may not be related with the timeout.
I printed the error message before timeout, it is actually `[Errno 13] Access denied`
```
Using drivers OS-X-MSC, libusb-MTP, libusb-vendor-specific
Looking for Sony devices

Querying mass storage device
Sony DSC is a camera in mass storage mode

Switching to app install mode
Waiting for camera to switch...
[Errno 13] Access denied (insufficient permissions)
[Errno 13] Access denied (insufficient permissions)
[Errno 13] Access denied (insufficient permissions)
```

I ran the install command again but with `sudo`, and the installation worked successfully!

**Comment by MasterMong** [comment]
I solved this problem by use USB3 instead of USB2.

**Comment by Daniel-Astete** [comment]
Hello guys, just to share my experience: I had a similar problem as @AndreMPCosta with my A7 IV that disconnected after trying to start the service mode, so I ended up running the Service Mode tweaking as fast as I could until it eventually worked. It was very frustrating but ended up working. 
Thanks!
